cli/src/commands/run.rs

```diff
@@ -24,24 +30,41 @@ use crate::ui::Ui;
 /// # Example
 ///
 /// # Run pre-commit on your local work
-/// $ jj run 'pre-commit.py .github/pre-commit.yaml' -r (main..@) -j 4
+/// $ jj run 'pre-commit run .github/pre-commit.yaml' -r (trunk()..@) -j 4
 ///
 /// This allows pre-commit integration and other funny stuff.
 #[derive(clap::Args, Clone, Debug)]
 #[command(verbatim_doc_comment)]
-pub(crate) struct RunArgs {
+pub struct RunArgs {
     /// The command to run across all selected revisions.
-    #[arg(long, short, alias = "x")]
-    command: String,
+    shell_command: String,
+    /// A no-op option to match the interface of `git rebase -x`.
+    #[arg(hide = true, alias = "x")]
+    unused_command: bool,
     /// The revisions to change.
+    /// Multiple revsets are accepted and the work will be done on a
+    /// intersection of them.
     #[arg(long, short, default_value = "@")]
     revisions: Vec<RevisionArg>,
+    /// How many processes should run in parallel, uses by default all cores.
+    #[arg(long, short)]
+    jobs: Option<usize>,
 }
 
-pub(crate) fn cmd_run(
-    _ui: &mut Ui,
-    _command: &CommandHelper,
-    _args: &RunArgs,
-) -> Result<(), CommandError> {
+pub fn cmd_run(ui: &mut Ui, command: &CommandHelper, args: &RunArgs) -> Result<(), CommandError> {
+    let workspace_command = command.workspace_helper(ui)?;
+    let _resolved_commits =
+        resolve_multiple_nonempty_revsets(&args.revisions, &workspace_command, ui)?;
+    let _jobs = if let Some(_jobs) = args.jobs {
+        _jobs
+    } else {
+        // Use all available cores
+
+        // SAFETY:
+        // We use a internal constant of 4 threads, if it fails
+        let available = std::thread::available_parallelism()
+            .unwrap_or(unsafe { NonZeroUsize::new_unchecked(4) });
+        available.into()
```

Comment:
Do you think we'll need parallelism/threads in the `jj run` itself? I would have thought that we would just run commands asynchronously and that a single thread for processing the results should be plenty. There must be async APIs for subprocessing, right?

Reply:
Yes, we'll need a upper limit in run internally. As Rust's futures do not map 1:1 to threads, so we'll limit in the process spawning type.

Reply:
I this `--jobs` should control how many subprocesses we run in parallel. I think how many is appropriate depends a lot on the command the user specified. If it's a slow but relatively light command (waiting on some server, perhaps), then it can be set to a high value. If it'a very heavy command (e.g. bazel), then their machine may only be able to handle one instance at a time. So I think it's better to simply default to the safe value of 1, so we don't run their system out of memory. What do you think?

Reply:
When #2517 lands, I'll change it to `repo.concurrency()` and then fallback to the amount of cores if it isn't available.  At this point the description should also change to something like "With which amount of internal parallelism it should run". 

Altough we explicitly say in the Design Doc, that we use the core count if no value is given.

Reply:
> When #2517 lands, I'll change it to `repo.concurrency()` and then fallback to the amount of cores if it isn't available.

I don't think the number of concurrent subprocesses (of the particular type you passed to `jj run` this time) your system can handle is very related to the optimal number of concurrent requests to a server-backed commit backend. But it seems like we keep misunderstanding each other, so it's probably a better use of both your time and mine to just merge this PR as is. It will be adjusted later anyway.